app/src/main/java/net/aiscope/gdd_app/ui/microscope_quality/MicroscopeQualityPresenter.kt

```diff
@@ -0,0 +1,37 @@
+package net.aiscope.gdd_app.ui.microscope_quality
+
+import android.content.Context
+import net.aiscope.gdd_app.repository.SampleRepository
+import timber.log.Timber
+import javax.inject.Inject
+
+class MicroscopeQualityPresenter @Inject constructor(
+    val view: MicroscopeQualityView,
+    private val mapper: MicroscopeQualityMapper,
+    val repository: SampleRepository,
+    private val context: Context
+) {
+
+    suspend fun showScreen() {
+        val lastMicroscopeQuality = repository.last()?.microscopeQuality
+        view.fillForm(mapper.convert(lastMicroscopeQuality))
+    }
+
+    suspend fun save(viewModel: MicroscopeQualityViewStateModel) {
+        val model = mapper.convert(viewModel)
+        try {
+            val sample = repository.current()
+                .copy(microscopeQuality = model)
+            repository.store(sample)
+            view.goToCaptureImage(sample.nextImageName())
+        } catch(@Suppress("TooGenericExceptionCaught") error: Throwable) {
```

Comment:
What errors can happen here? For me, the `TooGenericExceptionCaught` is a smell of something bad that might happen which might just cause the application to misbehave instead of letting it crash, so we can get a Crashlytics report with the issue.

Reply:
I agree - I just made it consistent with what's done in the Metadata screen... We should maybe raise a ticket to review these points?

Reply:
Created Trello card https://trello.com/c/HVEezUXD. Thanks!